utils_cv/detection/dataset.py

```diff
@@ -20,12 +20,48 @@
 from .bbox import AnnotationBbox
 from .mask import binarise_mask
 from .references.utils import collate_fn
-from .references.transforms import RandomHorizontalFlip, Compose, ToTensor
+from .references.transforms import Compose, ToTensor
 from ..common.gpu import db_num_workers
 
 Trans = Callable[[object, dict], Tuple[object, dict]]
 
 
+def _flip_keypoints(kps, width, hflip_inds):
+    """ Variation of `references.transforms._flip_coco_person_keypoints` with additional
+    hflip_inds. """
```

Comment:
See my comment in the notebook. Do we really need hflip_inds? And if not, does that mean that this function will not be needed anymore?

Reply:
In `utils_cv.detection.dataset.get_transform()`, `RandomHorizontalFlip` is used to flip the image, so `hflip_inds` is necessary.

Reply:
In training, the data would be augmented
https://github.com/microsoft/computervision-recipes/blob/8602f5ce79e5466ab15c1a69f17f82c5c87bd0fb/utils_cv/detection/dataset.py#L511-L512
by some kind of transformations:
https://github.com/microsoft/computervision-recipes/blob/8602f5ce79e5466ab15c1a69f17f82c5c87bd0fb/utils_cv/detection/dataset.py#L49-L79
`RandomHorizontalFlip` is one of the transformations used in training:
https://github.com/microsoft/computervision-recipes/blob/8602f5ce79e5466ab15c1a69f17f82c5c87bd0fb/utils_cv/detection/references/transforms.py#L26-L43
For generating flipped keypoint image data:
https://github.com/microsoft/computervision-recipes/blob/8602f5ce79e5466ab15c1a69f17f82c5c87bd0fb/utils_cv/detection/references/transforms.py#L6-L13
'left' would become 'right' after flipped horizontally.  Therefore, to get `DetectionDataset`work properly for keypoint detection in its `__getitem__()` method
https://github.com/microsoft/computervision-recipes/blob/8602f5ce79e5466ab15c1a69f17f82c5c87bd0fb/utils_cv/detection/dataset.py#L468
we need `hflip_inds` to indicate how to transform the keypoint data correctly.

Anyway, we can also leave it to users to define how to transform the data by themselves.

Reply:
I understand, thanks for the explanation. Since hflip_inds is a bit tricky to get right, would it be possible to allow it to be None (by default) and in that case the RandomHorizontalFlip() would throw an error with message something like "To use random horizontal flipping, hflip_inds needs to be specified"

Reply:
Done